### PR TITLE
Upgrade substrate to v0.9.25

### PR DESCRIPTION
**Pull request by vovacha**
- Pallets
  - Logic
  - Vec replaces with BoundedVec 
- Tests
- Benchmarks

### CODE REVIEW

File: flow/src/mock.rs [hunk 1]
```diff
@@ -4,33 +4,46 @@ pub use super::*;
...
 
-impl Campaign<Hash, AccountId, Balance, BlockNumber, Moment> {
-	pub fn new(campaign_id: Hash, expiry: BlockNumber) -> Campaign<Hash, AccountId, Balance, BlockNumber, Moment> {
+impl Campaign<Hash, AccountId, Balance, BlockNumber, Moment, BoundedVec<u8, <Test as Config>::StringLimit>> {
+	pub fn new(campaign_id: Hash, expiry: BlockNumber) -> Campaign<Hash, AccountId, Balance, BlockNumber, Moment, BoundedVec<u8, <Test as Config>::StringLimit>> {
 		Campaign {
 			id: campaign_id,
 			org: H256::random(),
-			name: vec![1, 2],
+			name: BoundedVec::truncate_from(vec![1, 2]),
```
Comment (2075) [inline flow/src/mock.rs:19]: afair we keep names off chain or do we need this for the slug/reservation later on?

File: flow/src/lib.rs [hunk 2]
```diff
@@ -535,24 +552,24 @@ pub mod pallet {
...
+			cid: BoundedVec<u8, T::StringLimit>,
+			token_symbol: BoundedVec<u8, T::StringLimit>, // up to 5
+			token_name: BoundedVec<u8, T::StringLimit>,   /* cleartext
 									* token_curve_a: u8,	  // preset
-									* token_curve_b: Vec<u8>, // custom */
+									* token_curve_b: BoundedVec<u8, T::StringLimit>, // custom */
 		) -> DispatchResultWithPostInfo {
 			let creator = ensure_signed(origin)?;
-			let owner = T::Control::org_controller_account(&org_id);
+			let owner = T::Control::org_controller_account(&org_id).ok_or(Error::<T>::InvalidId)?;
```
Comment (2075) [inline flow/src/lib.rs:568]: Could be the same Error as in L675, bu timho either both are invalid or unknown, correct?

**Review by 2075 — APPROVED** [review]
LGTM, thanks

File: flow/src/lib.rs (hunk 2 above)
Comment (vasylenko-yevhen) [reply flow/src/lib.rs:568]: Makes sense to return another error here. We already have `OwnerUnknown` one, gonna use that here.

File: flow/src/mock.rs (hunk 1 above)
Comment (vasylenko-yevhen) [reply flow/src/mock.rs:19]: True, name is stored off chain. It will be dropped in next updates. For now it exists for backwards compatibility.

**Pull request merged**